src/main/java/com/ibm/devops/notification/OTCAPIHelper.java

```diff
@@ -0,0 +1,249 @@
+/*
+ <notice>
+
+ Copyright 2017 IBM Corporation
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+ </notice>
+ */
+
+package com.ibm.devops.notification;
+
+
+import com.ibm.devops.dra.Util;
+import hudson.EnvVars;
+import hudson.ProxyConfiguration;
+import hudson.model.*;
+import hudson.security.ACL;
+import hudson.tasks.Recorder;
+import hudson.util.ListBoxModel;
+import hudson.util.Secret;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.apache.http.HttpHost;
+import org.apache.http.client.config.RequestConfig;
+import org.cloudfoundry.client.lib.CloudCredentials;
+import org.cloudfoundry.client.lib.CloudFoundryClient;
+import org.cloudfoundry.client.lib.HttpProxyConfiguration;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.regex.Pattern;
+import java.util.Scanner;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+
+/**
+ * OTCAPIHelper class helps collect various information (org, space, app) from OTC API
+ */
+public abstract class OTCAPIHelper extends Recorder {
+	// API URLs
+    private final static String ORGS_URL= "/v2/organizations?q=name:";
+    private final static String SPACES_URL= "/v2/spaces?q=name:";
+    private final static String APPS_URL= "/v2/apps?q=name:";
+    private final static String ORG= "&&organization_guid:";
+    private final static String SPACE= "&&space_guid:";
+    
+
+    public static void printPluginVersion(ClassLoader loader, PrintStream printStream) {
+        final Properties properties = new Properties();
```

Comment:
is there any specify reason that you have to create this file instead of reuse the AbstractDevOpsAction.java since most of the functions here are the same?